### DISCUSSION TITLE
CheckIf with multiple conditions / operator

### DISCUSSION THREAD

**Discussion by iamadlx** [GetPublii/Publii#1338] [Help] [answered]
I'm looking for an answer to this question:
https://forum.getpublii.com/topic/checkifany-helper-for-multiple-id-tag/

I tried this :
`{{#each tags}}
  {{#checkIf id 'notContains' [1,2,3]}}<a href="{{url}}" class="filter__item">{{name}}</a>{{/checkIf}}
  {{/each}}`

But the result is full of "false" and not the tags

Thank you in advance

**Comment by dziudek** [comment]
Hi,

First of all you should reverse arguments, because you must check if array [1,2,3] not contains ID:

```
{{#checkIf [1,2,3] 'notContains' id}}<a href="{{url}}" class="filter__item">{{name}}</a>{{/checkIf}}
```

**Reply by iamadlx** [reply]
Hi @dziudek !

Thank you for you feedback

`<nav class="filter">
                     {{#if @config.custom.tagFilterHome}}
                        <a href="{{@website.url}}" class="filter__item is-active">{{@config.custom.tagFilterHome}}</a>
                     {{/if}}
                     {{#each tags}}
                        {{#checkIf [1,2,3] 'notContains' id}}<a href="{{url}}" class="filter__item">{{name}}</a>{{/checkIf}}
                     {{/each}}
                  </nav>`

The problem with this is that I have this on output:
![image](https://github.com/GetPublii/Publii/assets/60820173/082d5e19-bd7a-4297-9325-ca00e747cb5f)

I'm trying to use the argument on the "source" theme in the tags to be displayed for sorting.

**Reply by dziudek** [reply]
@heyadlx - could you attach the whole hbs file? Then I would be able to try test this case - I supspect that there is an issue with context, but I cannot confirm it without whole hbs file

**Reply by iamadlx** [reply]
Here it is :)
https://we.tl/t-c8u9T21wMz

Nothing fancy, I just want to display the tags I want in the filter using its ID.

**Comment by dziudek** [comment, marked as answer]
@heyadlx - I know what happens - handlebars have specific handling of arrays/objects what leads to issues with passing arrays as arguments. Unfortunately it is connected strictly with hbs - that's why my unit tests was working fine, but in terms of hbs syntax it is a problem :/ 

In this case, the only solution is to fix these helpers in v.0.45

**Comment by dziudek** [comment]
@heyadlx - solved in https://github.com/GetPublii/Publii/issues/1345
